### ISSUE TITLE
Expose `storage` as public property on `SessionData`

### ISSUE THREAD

**Issue by benjaminmayo** [vapor/vapor#2480]
I am implementing a custom session driver backed by Redis. Whilst it is possible, I have to traffic through `Codable` to encode a JSON data blob as `SessionData` does not expose its storage (the `[String : String]` dictionary) as a public property. It is currently marked `internal`. If it was public, I could implement the `readSession` and `updateSession` methods more efficiently, by storing a string-string hash map directly in Redis for the given `SessionID`.

For a code path that is called so frequently, it would be nice if I could make it as efficient as possible.

**Comment by 0xTim** [comment]
Related to vapor/redis#162 - this will probably be needed when the Vapor sessions stuff is integrated with Redis

**Comment by Mordil** [comment]
We might not need to expose the whole property, we could just add ways to access it as if we exposed it, like a way to iterate through it.

**Comment by Mordil** [comment]
The way I've implemented sessions is that we provide a "good" default of using `Codable` and storing the session data as JSON, but someone could provide their own implementation without having to entirely implement `SessionDriver` themselves:

```swift
struct MyCustomRedisSessionsDelegate: RedisSessionsDelegate {
  func redis<Client: RedisClient>(_ client: Client, storeData data: SessionData, forID id: SessionID) -> EventLoopFuture<Void> {
    // your chosen method of storing the data
  }

  func redis<Client: RedisClient>(_ client: Client, fetchDataForID id: SessionID) -> EventLoopFuture<SessionData?> {
    // your chosen method of retrieving the data
  }
}
```

**Comment by 0xTim** [comment]
@benjaminmayo Does this satisfy your needs?

**Comment by benjaminmayo** [comment]
Is the access level of the `storage` property on SessionData changed to public?  I want to avoid Codable overhead.

**Comment by benjaminmayo** [comment]
For some additional context, when I was initially coding my sessions implementation I wanted to call through to hset/hget Redis operations directly, but I couldn’t because the storage was private.

**Comment by Mordil** [comment]
@0xTim What's being requested is orthogonal for the Sessions support that Redis provides. For someone to write their own implementation, aside from just storing it as JSON with Codable overhead as @benjaminmayo is talking about - we need to make changes to `SessionData`:

Either we (pseudo code to follow):
1. Provide a way to iterate through the keys, so subscript access can happen
```swift
var resultMap: [(String: Value)]
sessionData.keys.forEach {
  resultMap.append($0, sessionData[$0])
}
```
2. Expose the data structure directly somehow
```swift
sessionData.storage.reduce(...)
```

**Comment by 0xTim** [comment]
Ah ok, I misread that then. If there's a justification to making `storage` public (which there definitely appears to be) I'm happy for a PR to expose it, if you want to submit it @benjaminmayo ?

**Comment by benjaminmayo** [comment]
My Vapor app is basically a hobby (it's my blog) so haven't had a chance to code on it recently but when I do I'll get a PR up here. Thanks.

**Comment by Mordil** [comment]
I’m planning on tackling this today

**Comment by Mordil** [comment]
This has been resolved with changes that are available in [4.36.0](https://github.com/vapor/vapor/releases/tag/4.36.0)
